dist/AureliaPlugin.js

```diff
@@ -1,146 +1,146 @@
-for (let c in configModuleNames)
-    configModules[c] = { name: configModuleNames[c], exports: ["configure"] };
-// developmentLogging has a pre-task that uses ConsoleAppender
-configModules['developmentLogging'].exports.push("ConsoleAppender");
-function getConfigModules(config) {
-    if (!config)
-        return undefined;
-    if (!Array.isArray(config))
-        config = [config];
-    // Expand "standard"
-    let i = config.indexOf("standard");
-    if (i >= 0)
-        config.splice(i, 1, "basic", "history", "router");
-    // Expand "basic"
-    i = config.indexOf("basic");
-    if (i >= 0)
-        config.splice(i, 1, "defaultBindingLanguage", "defaultResources", "eventAggregator");
-    return config.map(c => configModules[c]);
-}
+"use strict";
+const AureliaDependenciesPlugin_1 = require("./AureliaDependenciesPlugin");
+const ConventionDependenciesPlugin_1 = require("./ConventionDependenciesPlugin");
+const DistPlugin_1 = require("./DistPlugin");
+const GlobDependenciesPlugin_1 = require("./GlobDependenciesPlugin");
+const HtmlDependenciesPlugin_1 = require("./HtmlDependenciesPlugin");
+const ModuleDependenciesPlugin_1 = require("./ModuleDependenciesPlugin");
+const PreserveExportsPlugin_1 = require("./PreserveExportsPlugin");
+const PreserveModuleNamePlugin_1 = require("./PreserveModuleNamePlugin");
+const SubFolderPlugin_1 = require("./SubFolderPlugin");
+class AureliaPlugin {
+    constructor(options = {}) {
+        this.options = Object.assign({
+            includeAll: false,
+            aureliaApp: "main",
+            aureliaConfig: ["standard", "developmentLogging"],
+            dist: "native-modules",
+            moduleMethods: [],
+            noHtmlLoader: false,
+            noModulePathResolve: false,
+            viewsFor: "src/**/*.{ts,js}",
+            viewsExtensions: ".html",
+        }, options);
+    }
+    apply(compiler) {
+        const opts = this.options;
+        // Make sure the loaders are easy to load at the root like `aurelia-webpack-plugin/html-resource-loader`
+        let resolveLoader = compiler.options.resolveLoader;
+        let alias = resolveLoader.alias || (resolveLoader.alias = {});
+        alias["aurelia-webpack-plugin"] = "aurelia-webpack-plugin/dist";
+        // Our async! loader is in fact just bundle-loader!.
+        alias["async"] = "bundle-loader";
+        if (opts.dist) {
+            // This plugin enables easy switching to a different module distribution (default for Aurelia is dist/commonjs).
+            let resolve = compiler.options.resolve;
+            let plugins = resolve.plugins || (resolve.plugins = []);
+            plugins.push(new DistPlugin_1.DistPlugin(opts.dist));
+        }
+        if (!opts.noModulePathResolve) {
+            // This plugin enables sub-path in modules that are not at the root (e.g. in a /dist folder),
+            // for example aurelia-chart/pie might resolve to aurelia-chart/dist/commonjs/pie

```

Comment:
It would be nice if we could move those comments out to a `/** */`-like documented TypeScript interface for options. Then we'd get the docs inline, when defining a `webpack.config.ts` file. Possibly something that can get in after we release this.